pyEpiabm/pyEpiabm/sweep/intervention_sweep.py

```diff
@@ -72,9 +86,20 @@ def __call__(self, time):
             num_cases = sum(map(lambda cell: cell.number_infectious(),
                             self._population.cells))
             if intervention.is_active(time, num_cases):
-                intervention(time)
                 if self.intervention_active_status[intervention] is False:
+                    # Update with the single intervention
+                    # (with respect to index)
+                    if hasattr(intervention, 'occurrence_index'):
+                        intervention_type = \
```

Comment:
This is very unwieldy - could the `intervention` object store its own type directly?

Reply:
This hasattr() function checks this active intervention has 'occurrence_index' which means it has several same type of interventions. We'll only update the parameter file when this intervention is active and its intervention type in the original parameter json file is stored as a list (a sequence of same type of interventions). When updating, we need its intervention_type (e.g. CaseIsolation etc.). So I think this batch of code is necessary, that I can't come up with a better way.

Reply:
> This hasattr() function checks this active intervention has 'occurrence_index' which means it has several same type of interventions. We'll only update the parameter file when this intervention is active and its intervention type in the original parameter json file is stored as a list (a sequence of same type of interventions). When updating, we need its intervention_type (e.g. CaseIsolation etc.). So I think this batch of code is necessary, that I can't come up with a better way.

So what is the difference between `intervention_type` and `type(intervention)`? I would have expected these to be the same (eliminating the complexity in L98-99), so if they're not then I think we need to think about renaming the `intervention_type` variable to something more accurate, or even replacing it with the type of the intervention object

Reply:
intervention_type is a string / key in the json file (e.g. 'place_closure', 'case_isolation), we use this to access / modify the value in parameter json file. type(intervention) is the pre-defined class, which we are trying to initialise (e.g. CaseIsolation(AbstractIntervention))

Reply:
Done!